### DISCUSSION TITLE
[JIT] darkMode: 'media' is always showing dark theme

### DISCUSSION THREAD

**Discussion by hannsadrian** [tailwindlabs/tailwindcss#4118] [Help]
This occurs to me while using tailwind with react and craco i. e. the postcss7 compatibility build. Tried it on multiple projects with the latest versions of everything.

**Comment by simonswiss** [comment]
Hey!

Are you able to share a minimal code repro of your setup in which this occurs?

**Reply by uncvrd** [reply]
@simonswiss hey! I do :) was just about to make a post about this!

Repo: https://github.com/uncvrd/tailwind-jit-dark-override

Video reproduction: https://streamable.com/uinuql

I have it set to show a background of **Red** for light theme and background of **Green** for dark mode. When I enable `jit` mode, the `dark` tags override any color tags. 

I'm using CRA with craco following the installation instructions in the docs. Let me know if there's anything else you need

**Reply by simonswiss** [reply]
Thanks so much for doing that example!

I can confirm that with `jit` enabled, the output from the `dark:bg-green-500` class is not wrapped into the necessary `@media (prefers-color-scheme: dark)` selector.

It's just being generated as `.dark\:bg-green-600`, and since it comes after the non-dark bg color in your example, it always overrides the red background.

I am off to bed now (it's quite late here in Australia), but we'll have a look into this 👍

**Reply by uncvrd** [reply]
No problem! Glad you were able to re-create on your end. This one has puzzled me for the past week haha
